js/ui/components/clubhouse.js

```diff
@@ -112,6 +112,50 @@ var ClubhouseNotificationSource = new Lang.Class({
     },
 });
 
+var ClubhouseButton = new Lang.Class({
+    Name: 'ClubhouseButton',
+    Extends: St.Button,
+
+    _init: function(clubhouseProxy) {
+        this.parent();
+        this._clubhouseProxy = clubhouseProxy;
+
+        this.add_style_class_name('clubhouse-shell-button');
+
+        this._icon_size = 64
+        let icon = new St.Icon({icon_name: 'user-home-symbolic',
+                                icon_size: this._icon_size});
+        this.add_actor(icon);
+        this.reposition();
+
+        Main.overview.connect('shown', () => {
+            this.updateVisibility();
+        });
+        Main.overview.connect('hidden', () => {
+            this.updateVisibility();
+        });
+        Main.sessionMode.connect('updated', () => {
+            this.updateVisibility();
+        });
+    },
+
+    reposition: function() {
+        let monitor = Main.layoutManager.primaryMonitor;
+        if (!monitor)
+            return;
+
+        this.x = monitor.width - Math.floor(this._icon_size / 2.0);
+        this.y = Math.floor(monitor.height / 2.0) - Math.floor(this.height / 2.0);
+    },
+
+    updateVisibility: function() {
+        let clubhouseShowing = this._clubhouseProxy && this._clubhouseProxy.g_name_owner &&
+            this._clubhouseProxy.Visible;
+        this.visible = !Main.sessionMode.isLocked && !Main.sessionMode.isGreeter &&
```

Comment:
The clubhouse component is disabled altogether by the `SessionMode` machinery when the session mode is switched to a locked or greeter session.
The way to handle it would be to call `removeChrome()` on the button, or destroy it, when the clubhouse component is disabled.